### PR TITLE
chore(pkg): Restrict `npm run spackle` to `prepublishOnly`

### PR DESCRIPTION
**Pull request by ExE-Boss**
The&nbsp;spackle script&nbsp;is, for&nbsp;whatever&nbsp;reason, run&nbsp;on&nbsp;all tested&nbsp;**Node.js**&nbsp;versions in&nbsp;CI.

This&nbsp;ensures&nbsp;that it&nbsp;at&nbsp;least doesn’t&nbsp;cause build&nbsp;failures.

### CODE REVIEW

**Comment by ljharb** [comment]
Ah, because i have it in prepublish. Simplest would be to use the in-publish module to gate it running only during publish.

File: operations/spackle.js [hunk 1]
```diff
@@ -5,7 +5,61 @@ var fs = require('fs');
...
+/** @type {(arg: any) => string} */
+var $ObjProtoToString = callBound('%Object.prototype.toString%');
+
+/** @type {(arg: any) => arg is (any[] | ReadonlyArray<any>)} */
+var $isArray = GetIntrinsic('%Array.isArray%', true)
+	|| function (arg) {
+		return $ObjProtoToString(arg) === '[object Array]';
+	};
+
+/** @type {<T>(array: T[], ...items: T[]) => number} */
```
Comment (ljharb) [inline operations/spackle.js]: Please do not add typescript-like comments to this repo; I’m not interested in the pollution.

File: operations/spackle.js [hunk 2]
```diff
@@ -5,7 +5,61 @@ var fs = require('fs');
 var childProcess = require('child_process');
 var deltas = require('./deltas');
 
-var writtenFiles = [5, 2015, 2016, 2017, 2018, 2019].flatMap((year, i, arr) => {
```
Comment (ljharb) [inline operations/spackle.js]: The change is not needed, but if it was, we’d use the array.prototype.flatmap package instead of inlining.

**Comment by ExE-Boss** [comment]
You&nbsp;might&nbsp;want to&nbsp;rename the&nbsp;deprecated&nbsp;`prepublish` to&nbsp;`prepublishOnly`&nbsp;then, that&nbsp;would&nbsp;resolve all&nbsp;your&nbsp;problems with&nbsp;`prepublish`‑only&nbsp;scripts running&nbsp;during the&nbsp;`prepare`&nbsp;stage.

**Comment by ljharb** [comment]
"prepublish" isn't ever going away, and "prepublishOnly" would allow older npm versions to publish incorrectly. I'll update this to use in-publish.

**Pull request merged**